modules/tinymce/src/core/main/ts/Rtc.ts

```diff
@@ -29,6 +29,11 @@ const runSerializerFiltersOnFragment = (editor: Editor, fragment: AstNode) => {
   FilterNode.filter(editor.serializer.getNodeFilters(), editor.serializer.getAttributeFilters(), fragment);
 };
 
+const getInsertContext = (editor: Editor) => {
+  const startElm = editor.selection.getStart(true);
+  return Type.isNullable(startElm) ? 'body' : startElm.nodeName.toLowerCase();
```

Comment:
Would this be better if no context was set if the start element is nullable? That's what we do in `SetSelectionContent.ts`, as `body` probably isn't an ideal fallback for inline mode for example.

Reply:
The RTC content model is a fun thing to wrap your head around, as far as it is concerned inline and iframe modes are the same. Any differences that matter will probably be taken care of in the rendering code.

As such I'm not sure it matters for RTC purposes what the context is with a null selection. We have to replicate `rng.createContextualFragment()` manually; what we really need is the raw unaltered HTML. We just don't want to build a new parser (yet). Without context the existing parser wraps plain text in a `p` tag which meant inserting simple things split the paragraph.

Actually, after thinking about that explanation and looking at `SetSelectionContent`, what we probably want is the same `{ isRootContent: true, forced_root_block: false }` settings used by `SetSelectionContent` instead of trying to calculate `context`? Does that makes sense @spocke?

Reply:
forced_root_block: false is something we should remove and let it be default for both enter and parsing since rtc will require it all to be paragraphs anyway and context should control how the auto wrapping or unwrapping should work to produce a valid fragment. Removing forced_root_block: false is something I've been meaning to add to v6 axe list. I think Lee is right we should just omit the context if we can't extract it since we do that elsewhere 99% of the time we will have a context anyway but it would be more consistent.